4337/test/eip4337/EIP4337ModuleExisting.spec.ts

```diff
@@ -12,14 +11,13 @@ import {
 import { chainId } from '../utils/encoding'
 
 describe('EIP4337Module - Existing Safe', async () => {
-  const [user1] = waffle.provider.getWallets()
+  const [user1] = await ethers.getSigners()
```

Comment:
When I was doing the migration for the safe-contracts repo, I had an issue with top-level async functions. I described it in this PR: https://github.com/safe-global/safe-contracts/pull/641

Quote:
> The migration showed some problems in how the tests were run. Some asynchronous functions were not properly accounted for. This was hard to spot if you run tests 1 by 1, but if you check out to the latest version and run hardhat test --parallel, then you can observe that the test suite finishes without waiting for all tests to finish
Here's an example of a problematic code block:
```
describe("SignMessageLib", async () => {
    const [user1, user2] = waffle.provider.getWallets();
```
> It looks like the test runner doesn't properly handle top-level async functions. As a solution, the wallet retrieval was moved into the setupTests function for each test, and the individual test can access it from the function's return values.

This was also confirmed by one of the hardhat maintainers here: https://github.com/NomicFoundation/hardhat/issues/3117#issuecomment-1235346851

This may be fine now, but we shall keep it in mind. It'd be great to address it in this PR if you have time.

Reply:
Oh, sneaky.

Yes, I can just move this to the setup function right? Will include in this PR.

Reply:
Yes, that's what I did in the Safe contracts repo

Reply:
Fixed.